target/linux/ath79/generic/base-files/etc/board.d/02_network

```diff
@@ -355,6 +359,10 @@ ath79_setup_macs()
 	devolo,magic-2-wifi)
 		label_mac=$(macaddr_add "$(mtd_get_mac_binary art 0x1002)" 3)
 		;;
+	dlink,dch-g020-a1)
+		lan_mac=$(mtd_get_mac_text "mp" 1)
+		ifconfig wlan0 hw ether $(mtd_get_mac_text "mp" 19)
```

Comment:
Wifi start is async. Your setup should go there:
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/generic/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac

Please have a look at the other devices for the correct setup.

And please replace 19 by 0x13

Reply:
Thanks, I initially tried to set it within `10-ath9k-eeprom` but eventually figured out that script was not even executed, apparently since the driver is happy with the `mtd-cal-data` from the dts.

Reply:
wmac and ath9k init are separate indeed

Reply:
> Please have a look at the other devices for the correct setup.

should I also use `macaddr_add` here?
Currently this is not used consistently between all boards (e.g. `siemens,ws-ap3610`), so I wonder what exactly it is doing?
However `[ "$PHYNBR" -eq 0 ] &&` is apparently not needed, as there is just one phy available.

Reply:
macaddr_add increments your address by the second argument. Based on your commit message, this is not necessary, as both addresses should be the same and LAN/Wifi are also meant to receive the same addresses.